GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -458,37 +458,98 @@ static void StraightenLanes(int startIndex, int lastIndex, IList<RevisionGraphRo
 
                     bool moved = false;
                     IRevisionGraphRow previousRow = localOrderedRowCache[currentIndex - 1];
-                    IRevisionGraphRow nextRow = localOrderedRowCache[currentIndex + 1];
+
                     foreach (RevisionGraphSegment revisionGraphSegment in currentRow.Segments)
                     {
                         int previousLane = previousRow.GetLaneIndexForSegment(revisionGraphSegment);
                         int currentLane = currentRow.GetLaneIndexForSegment(revisionGraphSegment);
-                        if (previousLane > currentLane)
+
+                        if (currentLane >= previousLane)
+                        {
+                            continue;
+                        }
+
+                        int straightenedCurrentLane = currentLane + 1;
+                        int lookaheadLane = currentLane;
+                        int nextIndex = currentIndex + 1;
+                        var segmentOrAncestor = localOrderedRowCache[currentIndex].FirstParentOrSelf(revisionGraphSegment);
+
+                        for (int lookaheadIndex = nextIndex; lookaheadLane == currentLane && lookaheadIndex <= Math.Min(currentIndex + _straightenLanesLookAhead, lastIndex); ++lookaheadIndex)
```

Comment:
in order to finish the elimination of `nextIndex`
```suggestion
                        for (int lookaheadIndex = currentIndex + 1; lookaheadLane == currentLane && lookaheadIndex <= Math.Min(currentIndex + _straightenLanesLookAhead, lastIndex); ++lookaheadIndex)
```